src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -619,23 +637,38 @@ - (BOOL)setupCommissioningSessionWithPayload:(MTRSetupPayload *)payload
 
         chip::NodeId nodeId = [newNodeID unsignedLongLongValue];
         self->_operationalCredentialsDelegate->SetDeviceID(nodeId);
-        auto errorCode = self->_cppCommissioner->EstablishPASEConnection(nodeId, [pairingCode UTF8String]);
+        errorCode = self->_cppCommissioner->EstablishPASEConnection(nodeId, [pairingCode UTF8String]);
         return ![MTRDeviceController checkForError:errorCode logMsg:kErrorPairDevice error:error];
     };
 
-    return [self syncRunOnWorkQueueWithBoolReturnValue:block error:error];
+    auto success = [self syncRunOnWorkQueueWithBoolReturnValue:block error:error];
+    if (!success) {
+        MATTER_LOG_METRIC_END(kMetricDarwinFrameworkDeviceCommissioning, errorCode);
```

Comment:
Why don't we log the metric on success as well?

Reply:
It is here: https://github.com/project-chip/connectedhomeip/pull/32582/files#r1525635022